### PR TITLE
Support u8 slice digester in frozen abi struct. 

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem
Support u8 slice digester in frozen abi struct. 

#### Summary of Changes

Fixes #

### CODE REVIEW

File: sdk/src/account.rs [hunk 1]
```diff
@@ -47,7 +47,7 @@ mod account_serialize {
         lamports: u64,
         #[serde(with = "serde_bytes")]
         // a ref so we don't have to make a copy just to serialize this
-        data: &'a Vec<u8>,
+        data: &'a [u8],
```
Comment (jeffwashington) [inline sdk/src/account.rs:50]: you can get rid of the 'data' parameter below and use account.data()

Comment (HaoranYi) [reply sdk/src/account.rs:50]: yup. fixed.

**Review by jeffwashington — APPROVED** [review]
lgtm

File: frozen-abi/Cargo.toml [hunk 2]
```diff
@@ -16,6 +16,7 @@ lazy_static = "1.4.0"
 log = "0.4.14"
 serde = "1.0.136"
 serde_derive = "1.0.103"
+serde_bytes = "0.11"
```
Comment (ryoqun) [inline frozen-abi/Cargo.toml:19]: (seems this should be under `dev-dependencies`...)

File: frozen-abi/src/abi_example.rs [hunk 3]
```diff
@@ -415,6 +415,13 @@ impl AbiExample for &Vec<u8> {
     }
 }
 
+impl AbiExample for &[u8] {
+    fn example() -> Self {
+        info!("AbiExample for (&[u8]): {}", type_name::<Self>());
+        &VEC_U8[..]
+    }
+}
+
```
Comment (ryoqun) [inline frozen-abi/src/abi_example.rs:424]: fyi, this impl will be replaced as par https://github.com/solana-labs/solana/pull/35118#discussion_r1494580278

**Pull request merged**